### DISCUSSION TITLE
Some Launchpad settings not being saved

### DISCUSSION THREAD

**Discussion by mayesp** [GrandOrgue/grandorgue#1348] [Help]
I was getting very cross when I kept losing Launchpad settings, and now having calmed down and analysed the situation, what I am seeing is very puzzling!

This is grandorgue-3.9.4-1.1.x86_64 running under SuSE Tumbleweed with latest updates. Sampleset is Bureå Extended AGO.

I have combinations set on all three manuals and pedals for the pistons, and eight generals. That works fine when operated by mouse and screen.

I have enabled GO to send colour settings back to the Launchpad for pistons, generals and couplers.

The couplers and generals work fine - the association with Launchpad buttons and the colours work fine, and are correctly saved between launches.

But when I set the Launchpad to be associated with the keyboard and pedalboard  pistons, they don't get saved, even though they work when first set.

Moreover, if I export the settings to another .cmb file, the keyboard and pedalboard settings are not there.

To give a bit more detail, the eight generals I have set are mapped to the 5th row up - buttons 48-55 - they work fine. Button 120 is general cancel, and the couplers are the RHS buttons 104, 88, 72, 56, 40 and 24 - again, they all work fine, including illuminating the Launchpad, and are saved.

But when I try to use the bottom four rows for pedal and three manual pistons - 112-119, 96-113, 80-87 and 64-71 and they don't get saved. 

Since I wasted a lot of time setting and then losing them, I have just tried setting the bottom left Launchpad button (112) to "Pedal 1". It works when I set it. But when I save settings, or export to a .cmb file, button 112 is not there, and if I exit and restart GO, the setting is lost. I even found an old saved file from a long time back that has

MIDIKey001=112
MIDISendKey001=112

in it, but I don't see that in the new automatically saved .cmb file, or the one that I manually export.

Sorry that is long-winded, but seems VERY confusing to me!!

Thanks in advance for any help.

**Comment by mayesp** [comment]
Additional information as requested:

Sample set is "Bureå Church (Extended AGO Version)"

First screenshot shows the setting for the "Pedal 1" piston - which I have assigned to the bottom left button - 112.

![Screenshot_20230121_161248](https://user-images.githubusercontent.com/9988235/213876514-0cf4f9fc-1071-4975-9e1a-1c66de7f6537.png)

Second screenshot shows the setting for sending the code to the Launchpad to illuminate the light. I had to select "alsa: Launchpad MIDI 1", "9x Note", "MIDI-note"=112 and "On value"=51 (bright amber)

![Screenshot_20230121_161409](https://user-images.githubusercontent.com/9988235/213876654-ae2830c1-379e-46bf-bc3a-8d33647ac0bf.png)

Third image shows that this setting works. Launchpad button works, and light illuminates whether I press Launchpad key or select with mouse

![IMG_20230121_161510180](https://user-images.githubusercontent.com/9988235/213876711-17770d0b-f0bb-42b0-a1d0-e1593415ef44.jpg)

I then "save", and also export the .cmb to another file. After a restart of GO, final screenshot shows that the setting have NOT been saved.

![Screenshot_20230121_161634](https://user-images.githubusercontent.com/9988235/213876763-8252f6fc-2047-41e4-ba31-6dd56f65c18f.png)

I have verified that
MIDIKey001=112
MIDISendKey001=112
is in a very old saved file (when all this worked), but NOT in either the current saved .cmb file, nor in the exported .cmb file.

Thanks -- Peter

**Comment by oleg68** [comment]
It is a bug of GrandOrgue specific for this sampleset. I created an issue #1350

**Reply by mayesp** [reply]
OK, thank you, I shall keep an eye open for updates.
